dsl/parser/ASTNode.py

```
@@ -160,8 +160,48 @@ def __generateParam(self, tokenList):
 
 class LayoutNode(ASTNode):
 
+    def __init__(self, list_of_tokens):
+        super().__init__(list_of_tokens)
+        self.rows = []
+
+    def parse(self):
+        while True:
+            if self.next().value == '[':
+                row = RowNode(self.tokens)
+                self.rows.append(row)
+                row.parse()
+            if self.peek().value == '}':
+                self.next()
+                break
+
+class RowNode(ASTNode):
+    def __init__(self, list_of_tokens):
+        super().__init__(list_of_tokens)
+        self.elements = []
+
+    def parse(self):
+        while True:
+            tk = self.peek()
+            if tk.is_a(Type.RESERVED):
+                node = ConstructorNode(self.tokens)
+                self.elements.append(node)
+                node.parse()
+            elif tk.is_a(Type.VARIABLE):
+                node = VarNode(self.tokens)
+                self.elements.append(node)
+                node.parse()

```

How about the case of String ?
[Nav,  "text", some_var]

---

Also, can we immediately throw exception if the token is obviously invalid ?
i.e. [1 2 3]
[= = =]
["]
etc.